### PR TITLE
fix: production mode support for concardis credit card

### PR DESCRIPTION
**Pull request by SGrueber**

### CODE REVIEW

File: src/app/pages/checkout-payment/components/payment-concardis-creditcard/payment-concardis-creditcard.component.ts [hunk 1]
```diff
@@ -186,33 +191,40 @@ export class PaymentConcardisCreditcardComponent implements OnInit, OnChanges, O
...
-
-      this.errorMessage.cvc = error.message.properties.find(prop => prop.key === 'verification');
-      if (this.errorMessage.cvc && this.errorMessage.cvc.code) {
-        this.errorMessage.cvc.messageKey = this.getErrorMessage(
-          this.errorMessage.cvc.code,
-          'cvc',
-          this.errorMessage.cvc.message
-        );
-      }
+      if (typeof error.message !== 'string' && error.message && error.message.properties) {
```
Comment (dhhyi) [inline src/app/pages/checkout-payment/components/payment-concardis-creditcard/payment-concardis-creditcard.component.ts]: I think when `typeof error.message !== 'string'` is true, then there has to be an object, so the following `error.message` is true at this point

File: src/app/core/services/script-loader/script-loader.service.ts [hunk 2]
```diff
@@ -12,16 +14,21 @@ export class ScriptLoaderService {
...
+      src: 'https://pp.payengine.de/bridge/1.0/payengine.min.js',
+      loaded: false,
+    },
   ];
 
   /**
    * load a script, if it has not already been loaded
    */
-  load(name: string): Observable<ScriptType> {
+  load(name: ScriptName): Observable<ScriptType> {
```
Comment (dhhyi) [inline src/app/core/services/script-loader/script-loader.service.ts]: this makes the `ScriptLoaderService` really special and makes further abstraction harder. I'd favor that the payment-concardis-creditcard.component knows which script to load and the service is more abstract

File: src/app/pages/checkout-payment/components/payment-concardis-creditcard/payment-concardis-creditcard.component.ts [hunk 3]
```diff
@@ -117,9 +117,14 @@ export class PaymentConcardisCreditcardComponent implements OnInit, OnChanges, O
         return;
       }
 
+      const environment =
```
Comment (dhhyi) [inline src/app/pages/checkout-payment/components/payment-concardis-creditcard/payment-concardis-creditcard.component.ts]: the name `environment` causes a lint warning

File: src/app/core/services/script-loader/script-loader.service.ts [hunk 4]
```diff
@@ -1,29 +1,29 @@
 import { Injectable } from '@angular/core';
 import { Observable, Observer } from 'rxjs';
 
+export type ScriptName = 'concardis-payengine-sandbox' | 'concardis-payengine-production';
```
Comment (dhhyi) [inline src/app/core/services/script-loader/script-loader.service.ts]: now this is unused :sweat_smile:

File: src/app/core/services/script-loader/script-loader.service.ts [hunk 5]
```diff
@@ -1,29 +1,29 @@
...
-    },
-  ];
+  private scripts: ScriptType[] = [];
 
   /**
    * load a script, if it has not already been loaded
+   * @param url  script url, e.g. https://pptest.payengine.de/bridge/1.0/payengine.min.js
    */
-  load(name: string): Observable<ScriptType> {
+  load(url: string): Observable<ScriptType> {
```
Comment (dhhyi) [inline src/app/core/services/script-loader/script-loader.service.ts]: Great. I like it :+1:

Comment (SGrueber) [reply src/app/core/services/script-loader/script-loader.service.ts]: Thx :)

**Pull request merged**